packages/ledger-icp/src/account_identifier.ts

```diff
@@ -97,12 +97,18 @@ export class SubAccount {
   }
 
   public static fromID(id: number): SubAccount {
-    if (id < 0 || id > 255) {
-      throw "Subaccount ID must be >= 0 and <= 255";
+    if (id < 0) throw new Error("Number cannot be negative");
+
+    if (id > Number.MAX_SAFE_INTEGER) {
+      throw new Error("Number is too large to fit in 32 bytes.");
     }
 
-    const bytes: Uint8Array = new Uint8Array(32).fill(0);
-    bytes[31] = id;
+    const buffer = Buffer.alloc(32);
```

Comment:
Looks good, thanks for the improvements! I've tested it, and it works well. I have just two questions, if you don't mind:

1. You are storing the unsigned 64-bit integer with `littleEndian` set to `false`. Did you double-check that this is correct?

2. In our utils function [bigIntToUint8Array](https://github.com/dfinity/ic-js/blob/main/packages/utils/src/utils/arrays.utils.ts#L15), we test if `setBigUint64` is present and, if not, we fall back on a manual calculation. While `setBigUint64` seems to be well-implemented across various platforms ([caniuse](https://caniuse.com/?search=setBigUint64)), I recall that we had an issue with Safari on mainnet about a year and a half ago (maybe @lmuntaner remembers as well). I was wondering if it might make sense to implement a fallback here too. What do you think?

I hope you don't mind my extra questions. The piece of code we are modifying is almost historic; I just want to be sure that everything will work well for everyone.

Reply:
I found this in our PRs: https://github.com/dfinity/agent-js/pull/577

There was some issue with setBigUint64 with old versions of iOS.

Reply:
It's probably worth implementing the fallback then? We could invest some times trying to figure out some metrics to see if it would make really sense but, if it is not too much work to implement a fallback and to follow the same pattern, maybe it "just" makes sense to implement the fallback.

Reply:
Thanks for the improvements and backwards compatible code, looks good to me.

Regarding big|littleendian. Final form uses the default `view.setBigUint64(24, BigInt(id));` which seem alright given that it's also the path with follow in the existing utils `bigIntToUint8Array`.